app/models/nic/base.rb

```diff
@@ -35,10 +35,15 @@ class Base < ActiveRecord::Base
     validate :exclusive_provision_interface
     validates :domain, :presence => true, :if => Proc.new { |nic| nic.host_managed? && nic.primary? }
     validate :valid_domain, :if => Proc.new { |nic| nic.host_managed? && nic.primary? }
-    validates :ip, :presence => true, :if => Proc.new { |nic| nic.host_managed? && nic.require_ip_validation? }
+    validates :ip, :presence => true, :if => Proc.new { |nic| nic.host_managed? && nic.require_ip4_validation? }
+    validates :ip6, :presence => true, :if => Proc.new { |nic| nic.host_managed? && nic.require_ip6_validation? }
 
-    validate :validate_host_location, :if => Proc.new { |nic| SETTINGS[:locations_enabled] && nic.subnet.present? }
-    validate :validate_host_organization, :if => Proc.new { |nic| SETTINGS[:organizations_enabled] && nic.subnet.present? }
+    validate :validate_subnet_types
+
+    validate :validate_host_subnet_location, :if => Proc.new { |nic| SETTINGS[:locations_enabled] && nic.host.present? && nic.subnet.present? }
```

Comment:
You can define these validations in terms of `Taxonomy.enabled_taxonomies`, so that these checks are not performed on every save.